### PR TITLE
Bump go 1.21 and k8s api libs

### PR DESCRIPTION
**Pull request by AKamyshnikova**
| Q               | A
| --------------- | ---
| Bug fix?        | []
| New feature?    | [x]
| API breaks?     | []
| Deprecations?   | []
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

Fixes: #152

### CODE REVIEW

**Comment by cscetbon** [comment]
> Fixes: #152 

Why go 1.21 and not 1.22 ?

**Comment by AKamyshnikova** [comment]
> > Fixes: #152
> 
> Why go 1.21 and not 1.22 ?

Some breaking changes in 1.22 and operator-sdk itself on 1.21 go https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L3

File: charts/casskop/crds/db.orange.com_cassandraclusters.yaml [hunk 1]
```diff
@@ -2184,7 +2187,7 @@ spec:
                               - type
                             properties:
                               localhostProfile:
-                                description: localhostProfile indicates a profile defined in a file on the node should be used. The profile must be preconfigured on the node to work. Must be a descending path, relative to the kubelet's configured seccomp profile location. Must only be set if type is "Localhost".
+                                description: localhostProfile indicates a profile defined in a file on the node should be used. The profile must be preconfigured on the node to work. Must be a descending path, relative to the kubelet's configured seccomp profile location. Must be set if type is "Localhost". Must NOT be set for any other type.
```
[the remark this answers is not in the document]
Comment (AKamyshnikova) [reply charts/casskop/crds/db.orange.com_cassandraclusters.yaml:2190]: These are autogenerated updates by `controller-gen crd` generation. If I revert them - not-missing-manifests will fail.

[the remark this answers is not in the document]
Comment (cscetbon) [reply charts/casskop/crds/db.orange.com_cassandraclusters.yaml:2190]: oh nvm I thought it was coming form our code

**Review by cscetbon — APPROVED** [review]

**Pull request merged**